### ISSUE TITLE
GIF animations with mixed disposal methods not handled correctly

### ISSUE THREAD

**Issue by kikokuo** [saucecontrol/PhotoSauce#114]
Hi, 
      I found some gif file leak image. I using  below code to handle flip and mirror. It worked great,
      But some gif file leak image when using library to process save or transfrom.
      How can I fix it or I miss some step?
```
PhotoSauce.MagicScaler.ProcessImageSettings processImageSettings = new()
{
	Width = gifPixelWidth,
	Height = gifPixelHeight,
	ResizeMode = PhotoSauce.MagicScaler.CropScaleMode.Stretch,
	HybridMode = PhotoSauce.MagicScaler.HybridScaleMode.Turbo
};

//Build pipeline And add orientation
using (var pl = MagicImageProcessor.BuildPipeline(m_sResourceFilePath, processImageSettings))
{
	using (var outfile = File.Create(sNewSourceFilePath))
	{
		if (IsVFlip)
			pl.AddTransform(new OrientationTransform(Orientation.FlipVertical));
		if (IsHFlip)
			pl.AddTransform(new OrientationTransform(Orientation.FlipHorizontal));				
		pl.WriteOutput(outfile);
	}
}

```
![VS_Originals_ad778d82-0272-422b-b9fe-96940fdb35a4_resource_download](https://github.com/saucecontrol/PhotoSauce/assets/39258415/84713acb-63c5-4ae0-8b69-dccbeb738b3c)
![image](https://github.com/saucecontrol/PhotoSauce/assets/39258415/033d9a9a-b93b-413f-b366-34a0e69ab915)

![gif save test - 2023070701](https://github.com/saucecontrol/PhotoSauce/assets/39258415/ac9f5841-a546-4b03-b975-337999a08309)
![image](https://github.com/saucecontrol/PhotoSauce/assets/39258415/8579422d-a6d0-4fcb-ac4c-57b2e9ec2d75)

**Comment by saucecontrol** [comment]
Thanks for the repro!  The sample input you've provided has an unusual mix of [frame disposal methods](https://www.oreilly.com/library/view/web-design-in/0596009879/ch32s04s03.html), which is not handled correctly in MagicScaler's animation optimizer.  I'll get a fix pushed for that soon.

**Comment by saucecontrol** [comment]
Thanks again for the report.  The issue was quite a bit more complicated than I expected (I had made some invalid assumptions in my initial implementation), but it's finally fixed in [11e96cc](https://github.com/saucecontrol/PhotoSauce/commit/11e96ccb8d9b4a26843ff3909aa03adda2679e50)
